tealium/core/consentmanager/ConsentManagerModule.swift

```diff
@@ -87,7 +87,12 @@ class ConsentManagerModule: DispatchValidator {
     }
 
     func getConsentData() -> [String: Any] {
-        if let consentDictionary = consentManager?.currentPolicy.policyTrackingData {
+        if var consentDictionary = consentManager?.currentPolicy.policyTrackingData {
+            let consentCategoriesKey = config.overrideConsentCategoriesKey
+            if consentCategoriesKey != nil {
+                consentDictionary[consentCategoriesKey!] = consentDictionary[TealiumDataKey.consentCategoriesKey]
+                consentDictionary.removeValue(forKey: TealiumDataKey.consentCategoriesKey)
```

Comment:
remove the value only if the two keys are different.
this could also just be a check on the preceding if like
`if let newKey = ..., newKey != oldKey {`